### PR TITLE
Add Client sync mode

### LINKED ISSUE #73: Sync mode(push pull, pushonly)
https://github.com/yorkie-team/yorkie-js-sdk/pull/495
https://github.com/yorkie-team/yorkie-js-sdk/pull/497

### PR DESCRIPTION
**Pull request by humdrum**
**What this PR does / why we need it**:
- Implement Client Sync mode 
  - In pushOnly mode,
    - Set pushOnly flag in the PushPull API.  
    - Don't apply server changes received by the push-pull response.
  - ~~In pullOnly mode,~~
    - ~~Don't send local changes.~~
    - ~~Don't send the sync request.~~
-  ~~Improve applyChangePack()~~
  - ~~If the push-pull response contains already performed changes. drop those changes.~~  

**Which issue(s) this PR fixes**:

Fixes #73 

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

**Additional documentation**:

```docs

```

**Checklist**:
- [ ] Added relevant tests or not required
- [ ] Didn't break anything

### CODE REVIEW

**Comment by humdrum** [comment]
I modified the code according to the implementation of JS.
- Remove pull-only mode
- Apply (https://github.com/yorkie-team/yorkie-js-sdk/pull/497)

@hackerwins

File: Examples/TextEditorApp/TextEditorApp/TextEditor/TextViewModel.swift [hunk 1]
```diff
@@ -75,10 +78,7 @@ class TextViewModel {
                 events.filter { $0.actor != clientID }.forEach {
                     switch $0.type {
                     case .content:
-                        let range = NSRange(location: $0.from, length: $0.to - $0.from)
-                        let content = $0.content ?? ""
-
-                        textChanges.append(.edit(range: range, content: content))
+                        break
```
Comment (myupage) [inline Examples/TextEditorApp/TextEditorApp/TextEditor/TextViewModel.swift:81]: Logic has been modified. Just double checking if this was done on purpose.

File: Sources/API/V1/yorkie/v1/yorkie.pb.swift [hunk 2]
```diff
@@ -1006,13 +1010,17 @@ extension Yorkie_V1_PushPullChangesRequest: SwiftProtobuf.Message, SwiftProtobuf
     try { if let v = self._changePack {
       try visitor.visitSingularMessageField(value: v, fieldNumber: 3)
     } }()
+    if self.pushOnly != false {
```
Comment (myupage) [inline Sources/API/V1/yorkie/v1/yorkie.pb.swift:1013]: Suggest changing it to 
```swift
if self.pushOnly  {
```
for better readability.

**Review by myupage — APPROVED** [review]

File: Sources/API/V1/yorkie/v1/yorkie.pb.swift (hunk 2 above)
Comment (humdrum) [reply Sources/API/V1/yorkie/v1/yorkie.pb.swift:1013]: I totally agreed. but This code was produced automatically by the protoc. So I can't change it. ^^;

File: Examples/TextEditorApp/TextEditorApp/TextEditor/TextViewModel.swift (hunk 1 above)
Comment (humdrum) [reply Examples/TextEditorApp/TextEditorApp/TextEditor/TextViewModel.swift:81]: That logic was replaced by syncText in line 105. Thanks.

File: Tests/Integration/ClientTests.swift [hunk 3]
```diff
@@ -155,4 +155,148 @@ class ClientTests: XCTestCase {
         XCTAssertFalse(isActive)
         XCTAssert(status == .deactivated)
     }
+
+    func test_sync_option_with_multiple_clients() async throws {
+        let rpcAddress = RPCAddress(host: "localhost", port: 8080)
```
[the remark this answers is not in the document]
Comment (humdrum) [reply Tests/Integration/ClientTests.swift:160]: I added test codes for pauseRemoteChanges and resumeRemoteChanges.

**Review by chacha912 — APPROVED** [review]

**Pull request merged**